auth.php

```diff
@@ -147,4 +192,100 @@ public function user_login ($username, $password) {
         return false;
     }
 
+    /**
+     * Check if the password is master password.
+     * @param $userpassword
+     * @return bool
+     */
+    private function is_master_password($userpassword) {
+        global $CFG;
+        if (isset($CFG->forced_plugin_settings)) {
```

Comment:
Do we need to store the master password as MD5 hash instead of plain text? Then we will need to provide user an option to generate hash.

Reply:
No, I think this is overkill. Remember the db password is in this same file, if you can read this file then all bets are off already